src/main/scala/viper/gobra/frontend/info/implementation/typing/ExprTyping.scala

```diff
@@ -492,6 +499,7 @@ trait ExprTyping extends BaseTyping { this: TypeInfoImpl =>
       case (VariadicT(elem), IntT(_)) => elem
       case (MapT(key, elem), indexT) if assignableTo(indexT, key) =>
         InternalSingleMulti(elem, InternalTupleT(Vector(elem, BooleanT)))
+      case (MathMapT(key, elem), indexT) if assignableTo(indexT, key) => elem
```

Comment:
Why is `InternalSingleMulti` not used here analogously to `MapT`? Isn't this required for safe map lookup?